Revert some of teamlist changes regarding inactive members/invites

zapu opened the pull request.

maxtaco commented on go/systests/team_list_test.go:
```diff
@@ -169,19 +169,32 @@ func TestTeamDuplicateUIDList(t *testing.T) {
+	_ = uvBefore
+	_ = uvAfter
+
+	// Commented out until we figure out the duplicate uvs story
```
please make a ticket and delete this code; include the commented out code in the ticket.

mmaxim commented on go/teams/service_helper.go:
```diff
@@ -87,6 +87,13 @@ func Details(ctx context.Context, g *libkb.GlobalContext, name string, forceRepo
 		if cat != keybase1.TeamInviteCategory_KEYBASE {
 			continue
 		}
+		if !invite.UserActive {
```
Do we need this? Isn't the map enough to de-dupe everything?

mmaxim commented:
This does appear to fix losers7

mmaxim commented:
Still see this on keybasefriends though:
![image](https://user-images.githubusercontent.com/1250314/33674932-09dc9e38-da7f-11e7-8138-a673a2c7666e.png)

zapu commented:
i'm not seeing anything more i can revert for this, it looks like a behavior that hasn't changed. I've added more dedup-ing to team.Details but I feel there are cases where it won't be correct

maxtaco commented:
This user is in the team as a reset user, and has an invite without an eldest seqno. It's likely a way deeper bug. Maybe we shouldn't do any more here and just fix it next release. I don't think the bug was introed by @zapu's PR

mmaxim commented:
FWIW the change to `AllUserVersions` that he made did fix my losers7 case, might be worth keeping. I think we can drop the commit he just put in though.

zapu commented:
lmk if I should drop the latest one though or we can clean all of this after release

mmaxim commented:
I think that commit makes it look like that reset user in keybasefriends is in the team, but he actually isn't. I would drop it.

zapu commented:
I dropped last commit and cleaned up the test. This leaves us with `losers7` with just one `candrencil` reported as *locked out*, however `keybasefriends` now has two `danebiletzer`s, one of which is *locked out*. This also has the effect of member counts being not consistent (member list reports 423 members, team list 422).

maxtaco commented:
thanks for that write-up. that agrees with my understanding. we'll have to figure out WTF next week.
